### PR TITLE
fix ints_to_bytes performance

### PR DESCRIPTION
**Pull request by mike-hunhoff**
Apparently `int.to_bytes` is horribly inefficient.

Before changes
```
        7   13.215    1.888   13.620    1.946 helpers.py:60(get_block_bytes)
```

After changes:
```
        7    0.000    0.000    0.044    0.006 helpers.py:58(get_block_bytes)
```

This PR also contains small API changes to the Ghidra helper functions.

### CODE REVIEW

File: capa/features/extractors/ghidra/helpers.py [hunk 1]
```diff
@@ -20,24 +20,25 @@
...
-    """Transform signed ints from Java into bytes for Python
+def ints_to_bytes(bytez: List[int]) -> bytes:
+    """convert Java signed ints to Python bytes
 
     args:
-        b: signed int returned from Java processing
+        bytez: list of Java signed ints
     """
-    return (b & 0xFF).to_bytes(1, "little")
+    return bytes([b & 0xFF for b in bytez])
```
Comment (mike-hunhoff) [inline capa/features/extractors/ghidra/helpers.py:29]: Apparently `int.to_bytes` is horrible inefficient.

File: capa/features/extractors/ghidra/helpers.py [hunk 2]
```diff
@@ -20,24 +20,25 @@
...
 
 
-def find_byte_sequence(seq: bytes) -> Iterator[int]:
+def find_byte_sequence(addr: ghidra.program.model.address.Address, seq: bytes) -> Iterator[int]:
```
Comment (mike-hunhoff) [inline capa/features/extractors/ghidra/helpers.py:32]: Makes `find_byte_sequence` generic by allowing caller to specify start address.

File: capa/features/extractors/ghidra/helpers.py [hunk 3]
```diff
@@ -20,24 +20,25 @@
...
     """yield all ea of a given byte sequence
 
     args:
+        addr: start address
         seq: bytes to search e.g. b"\x01\x03"
     """
     seqstr = "".join([f"\\x{b:02x}" for b in seq])
-    # .add(1) to avoid false positives on regular PE files
-    eas = findBytes(currentProgram().getMinAddress().add(1), seqstr, java.lang.Integer.MAX_VALUE, 1)  # type: ignore [name-defined] # noqa: F821
+    eas = findBytes(addr, seqstr, java.lang.Integer.MAX_VALUE, 1)  # type: ignore [name-defined] # noqa: F821
```
Comment (mike-hunhoff) [inline capa/features/extractors/ghidra/helpers.py:40]: See comment above.

File: capa/ghidra/helpers.py [hunk 4]
```diff
@@ -42,31 +43,23 @@ def seek(self, offset, whence=0):
...
-            bytez = bytez + capa.features.extractors.ghidra.helpers.fix_byte(fbytes.getOriginalByte(i))
-        return bytez
+    def get_bytes(self):
+        file_bytes = currentProgram().getMemory().getAllFileBytes()[0]  # type: ignore [name-defined] # noqa: F821
+
+        # getOriginalByte() allows for raw file parsing on the Ghidra side
+        # other functions will fail as Ghidra will think that it's reading uninitialized memory
+        bytes_ = [file_bytes.getOriginalByte(i) for i in range(file_bytes.getSize())]
+
+        return capa.features.extractors.ghidra.helpers.ints_to_bytes(bytes_)
```
Comment (mike-hunhoff) [inline capa/ghidra/helpers.py:62]: Read all and fix all file bytes once, to avoid unnecessary overhead later.

**Review by colton-gabertan — APPROVED** [review]
Awesome refactoring & performance update. I'm sure that dramatically reduces runtime! Thanks @mike-hunhoff

**Pull request merged**